x-pack/test/alerting_api_integration/spaces_only/tests/alerting/flapping_history.ts

```diff
@@ -0,0 +1,232 @@
+      const pattern = {
+        instance: [true, true, true, true],
+      };
+
+      const alertId = await createAlert(pattern, actionId);
+      objectRemover.add(space.id, alertId, 'rule', 'alerting');
+
+      let state = await getRuleState(start, alertId);
+      expect(get(state, ACTIVE_PATH)).to.eql([true]);
+      expect(state.alertRecoveredInstances).to.eql({});
+
+      start = new Date().toISOString();
+      state = await getRuleState(start, alertId, 2, true);
+      expect(get(state, ACTIVE_PATH)).to.eql([true, false]);
+      expect(state.alertRecoveredInstances).to.eql({});
+
+      start = new Date().toISOString();
+      state = await getRuleState(start, alertId, 3, true);
+      expect(get(state, ACTIVE_PATH)).to.eql([true, false, false]);
+      expect(state.alertRecoveredInstances).to.eql({});
+
+      start = new Date().toISOString();
+      state = await getRuleState(start, alertId, 4, true);
+      expect(get(state, ACTIVE_PATH)).to.eql([true, false, false, false]);
+      expect(state.alertRecoveredInstances).to.eql({});
+    });
+
+    it('should update flappingHistory when the alert remains recovered', async () => {
+      let start = new Date().toISOString();
+      const pattern = {
+        instance: [true, false, false, false, true],
+      };
+
+      const alertId = await createAlert(pattern, actionId);
+      objectRemover.add(space.id, alertId, 'rule', 'alerting');
+
+      let state = await getRuleState(start, alertId);
+      expect(get(state, ACTIVE_PATH)).to.eql([true]);
+      expect(state.alertRecoveredInstances).to.eql({});
+
+      start = new Date().toISOString();
+      state = await getRuleState(start, alertId, 2, true, true);
+      expect(get(state, RECOVERED_PATH)).to.eql([true, true]);
+      expect(state.alertInstances).to.eql({});
+
+      start = new Date().toISOString();
+      state = await getRuleState(start, alertId, 3, true, true);
+      expect(get(state, RECOVERED_PATH)).to.eql([true, true, false]);
+      expect(state.alertInstances).to.eql({});
+
+      start = new Date().toISOString();
+      state = await getRuleState(start, alertId, 4, true, true);
+      expect(get(state, RECOVERED_PATH)).to.eql([true, true, false, false]);
+      expect(state.alertInstances).to.eql({});
+
+      start = new Date().toISOString();
+      state = await getRuleState(start, alertId, 5, true, false);
+      expect(get(state, ACTIVE_PATH)).to.eql([true, true, false, false, true]);
+      expect(state.alertRecoveredInstances).to.eql({});
+    });
+  });
+
+  async function getState(start: string, runs: number, recovered: boolean) {
+    const result: any = await retry.try(async () => {
+      const searchResult = await es.search({
+        index: '.kibana_task_manager',
+        body: {
+          query: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    'task.taskType': 'alerting:test.patternFiring',
```

Comment:
Seems like we should use the task ID here and not task type.  But I don't think you can do a search with the id, you have to do a get on the doc, which is probably not what you want, given the range quantifier.  

The problem is that if some other test uses this rule type and forgets to delete their rule, the search may return that.  More likely case - you ctrl-c out of FTR, which ends up leaving a rule running, then you run FTR again and that rule is still running :-)

I guess having a check that the search results below include the expected id (which it will), that would be a good enough check ...

I wouldn't spend a lot of time on this, not a biggie.

Reply:
I think you are right that it would return more than 1 doc, but I think I resolved it here https://github.com/elastic/kibana/pull/144415/commits/bd748cd1fa3fdbcb3cdd8689b593a9088afd6756